scripts/Calibration/tofpd/diagnostics.py

```diff
@@ -575,3 +575,44 @@ def plot_peakd(wksp: Union[str, Workspace2D], peak_positions: Union[float, list]
     plt.show()
 
     return fig, ax
+
+def plot_corr(tof_ws):
+    """
+    Plots Pearson correlation coefficient for each detector
+    :param tof_ws: Workspace returned from collect_fit_result
+    :return: plot, plot axes
+    """
+    if not mtd.doesExist(str(tof_ws)):
+        raise ValueError("Could not find the provided workspace in ADS")
+
+    tof_ws = mtd[str(tof_ws)]
+
+    detTOF = tof_ws.getNumberHistograms()
+
+    fig, ax = plt.subplots()
+    ax.set_xlabel("det IDs")
+    ax.set_ylabel("Pearson correlation coefficient (TOF, d)")
+
+    r_vals = []
```

Comment:
Since you already know the length, `r_vals` and `detID` could both be created with
```python
r_vals = np.empty((numHist,), dtype=float)
r_vals.fill(np.nan)
```